tests/components/zeversolar/test_init.py

```diff
@@ -27,34 +26,21 @@ async def test_async_setup_entry_fails(
         pytest.raises(ConfigEntryNotReady),
     ):
         await init.async_setup_entry(hass, config_entry)
-
+        assert config_entry.state is ConfigEntryState.SETUP_RETRY
 
     with (
         patch("homeassistant.components.zeversolar.PLATFORMS", []),
         patch("zeversolar.ZeverSolarClient.get_data", return_value=zeversolar_data),
     ):
-        config_entry.add_to_hass(hass)
-
         result = await hass.config_entries.async_setup(config_entry.entry_id)
         await hass.async_block_till_done()
```

Comment:
Since the entry is in state `SETUP_RETRY` you need to use `async_schedule_reload` (and there is no need to block until done): 
```suggestion
        hass.config_entries.async_schedule_reload(config_entry.entry_id)
```